installer/main_darwin.go

```diff
@@ -49,36 +51,36 @@ func findExecutable(ctx context.Context, defaultOnly bool) string {
 	return ""
 }
 
-func installOllama(ctx context.Context, release, executablePath string) (string, error) {
+func installOllama(ctx context.Context, release, installPath string) (string, error) {
+	succeeded := false
+	executablePath := filepath.Join(installPath, "ollama")
+
 	if _, err := os.Stat(executablePath); err == nil {
 		return executablePath, nil
 	} else if !errors.Is(err, os.ErrNotExist) {
 		return "", fmt.Errorf("failed to check ollama executable: %w", err)
 	}
 
-	assetURL, err := getReleaseAssetURL(ctx, release, "ollama-darwin")
-	if err != nil {
-		return "", err
-	}
-
-	log.Printf("Downloading ollama from %s...", assetURL)
+	defer func() {
+		if !succeeded {
+			// On failure, remove partially extracted files.
+			_ = os.RemoveAll(installPath)
+		}
+	}()
 
-	// For darwin, Ollama is a single executable.
-	if err = os.MkdirAll(filepath.Dir(executablePath), 0o755); err != nil {
+	if err := os.MkdirAll(installPath, 0o755); err != nil {
 		return "", fmt.Errorf("failed to create ollama directory: %w", err)
 	}
-	file, err := os.OpenFile(executablePath, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0o755)
+
+	assetURL, err := getReleaseAssetURL(ctx, release, "ollama-darwin.tgz")
 	if err != nil {
-		return "", fmt.Errorf("failed to create executable: %w", err)
+		return "", err
 	}
-	defer file.Close()
-	succeeded := false
-	defer func() {
-		if !succeeded {
-			os.Remove(executablePath)
-		}
-	}()
 
+	log.Printf("Downloading ollama from %s...", assetURL)
+
+	// For Linux, Ollama is an archive that we need to extract.
```

Comment:
Pretty sure this is not Linux.

Reply:
I agree :). I changed the comment.

Reply:
> Consider refactoring this into a shared function

I will look into this after 1.18.1 release.